File: CS/Solution28.Module/BusinessObjects/DomainObject1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Xpo;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Base;

namespace Solution28.Module.BusinessObjects {
    [DefaultClassOptions]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session) : base(session) { }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
        private DateTime _Date;
        public DateTime Date {
            get { return _Date; }
            set { SetPropertyValue("Date", ref _Date, value); }
        }
    }

}

File: VB/Solution28.Module/BusinessObjects/DomainObject1.vb
Imports DevExpress.Xpo
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Base

Namespace Solution28.Module.BusinessObjects

    <DefaultClassOptions>
    Public Class DomainObject1
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private _Name As String

        Public Property Name As String
            Get
                Return _Name
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Name", _Name, value)
            End Set
        End Property

        Private _Date As Date

        Public Property [Date] As Date
            Get
                Return _Date
            End Get

            Set(ByVal value As Date)
                SetPropertyValue("Date", _Date, value)
            End Set
        End Property
    End Class
End Namespace
